presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveQueries.java

```diff
@@ -596,6 +596,9 @@ public void testUnionAll()
         assertQuery("SELECT distinct orderkey FROM (" +
                 "SELECT orderkey FROM lineitem WHERE linenumber = 5 " +
                 "UNION ALL SELECT orderkey FROM lineitem WHERE linenumber = 6)");
+
+        assertQuery("WITH t AS (SELECT null as a, null as b UNION ALL SELECT 'xxx' as a, 12 as b) " +
+                "SELECT * FROM t, t as u WHERE t.a = u.a and t.b = u.b");
```

Comment:
How does this trigger the sources with the different layout, I don't understand?

Reply:
For this query, Presto generates a plan with the following fragment:

```
 Fragment 1 [HASH]
     Output layout: [expr_9, expr_10]
     Output partitioning: SINGLE []
     Stage Execution Strategy: UNGROUPED_EXECUTION
     - InnerJoin[("expr_9" = "expr_27") AND ("expr_10" = "expr_28")][$hashvalue, $hashvalue_55] => [expr_9:varchar(3), expr_10:integer]
             Estimates: {rows: 1 (54B), cpu: 858.00, memory: 84.00, network: 168.00}
             Distribution: PARTITIONED
         - LocalExchange[ROUND_ROBIN] () => [expr_9:varchar(3), expr_10:integer, $hashvalue:bigint]
                 Estimates: {rows: 2 (84B), cpu: 318.00, memory: 0.00, network: 84.00}
             - RemoteSource[2] => [expr_3:varchar(3), expr_4:integer, $hashvalue_51:bigint]
             - RemoteSource[3] => [expr_5:varchar(3), expr_6:integer, $hashvalue_53:bigint]
         - LocalExchange[HASH][$hashvalue_55] (expr_27, expr_28) => [expr_27:varchar(3), expr_28:integer, $hashvalue_55:bigint]
                 Estimates: {rows: 2 (84B), cpu: 318.00, memory: 0.00, network: 84.00}
             - RemoteSource[4] => [expr_22:integer, expr_21:varchar(3), $hashvalue_56:bigint]
             - RemoteSource[5] => [expr_23:varchar(3), expr_24:integer, $hashvalue_58:bigint]
```

Here, the LocalExchange[HASH] plan node has 2 sources with very different output layouts:

```
         - LocalExchange[HASH][$hashvalue_55] (expr_27, expr_28) => [expr_27:varchar(3), expr_28:integer, $hashvalue_55:bigint]
                 Estimates: {rows: 2 (84B), cpu: 318.00, memory: 0.00, network: 84.00}
             - RemoteSource[4] => [expr_22:integer, expr_21:varchar(3), $hashvalue_56:bigint]
             - RemoteSource[5] => [expr_23:varchar(3), expr_24:integer, $hashvalue_58:bigint]
```